analyzers/tests/SonarAnalyzer.UnitTest/Common/FixAllProviders/DocumentBasedFixAllProviderTest.cs

```diff
@@ -0,0 +1,124 @@
+
+            document1BeforeContent.Should().NotBe(document1AfterContent);
+
+            var document2BeforeRoot = await document2Before.GetSyntaxRootAsync();
+            var document2BeforeContent = document2BeforeRoot.GetText().ToString();
+            var document2AfterRoot = await document2After.GetSyntaxRootAsync();
+            var document2AfterContent = document2AfterRoot.GetText().ToString();
+
+            document2BeforeContent.Should().Be(document2AfterContent);
+        }
+
+        [DataTestMethod]
+        [DataRow(FixAllScope.Project)]
+        [DataRow(FixAllScope.Solution)]
+        public async Task GetFixAsync_ForProjectAndSolution_AllFilesAreFixed(FixAllScope scope)
+        {
+            var codeFix = new DummyCodeFixCS();
+            var project = CreateProject();
+            var document1Before = project.FindDocument(Path.GetFileName("MyFile1.cs"));
+            var document2Before = project.FindDocument(Path.GetFileName("MyFile2.cs"));
+            var compilation = project.GetCompilation();
+            var diagnostics = DiagnosticVerifier.GetDiagnosticsNoExceptions(compilation, new DummyAnalyzerCS(), CompilationErrorBehavior.Ignore).ToImmutableArray();
+
+            var fixAllContext = new FixAllContext(project.Project, codeFix, scope, "Dummy Action", codeFix.FixableDiagnosticIds, new FixAllDiagnosticProvider(diagnostics), default);
+            var result = await SonarAnalyzer.Common.DocumentBasedFixAllProvider.Instance.GetFixAsync(fixAllContext);
+            var executedOperation = await result.GetOperationsAsync(default);
+            var document1After = executedOperation.OfType<ApplyChangesOperation>().Single().ChangedSolution.GetDocument(document1Before.Id);
+            var document2After = executedOperation.OfType<ApplyChangesOperation>().Single().ChangedSolution.GetDocument(document2Before.Id);
+
+            var document1BeforeRoot = await document1Before.GetSyntaxRootAsync();
+            var document1BeforeContent = document1BeforeRoot.GetText().ToString();
+            var document1AfterRoot = await document1After.GetSyntaxRootAsync();
+            var document1AfterContent = document1AfterRoot.GetText().ToString();
+
+            document1BeforeContent.Should().NotBe(document1AfterContent);
+
+            var document2BeforeRoot = await document2Before.GetSyntaxRootAsync();
+            var document2BeforeContent = document2BeforeRoot.GetText().ToString();
+            var document2AfterRoot = await document2After.GetSyntaxRootAsync();
+            var document2AfterContent = document2AfterRoot.GetText().ToString();
+
+            document2BeforeContent.Should().NotBe(document2AfterContent);
+        }
+
+        private static ProjectBuilder CreateProject() =>
+            SolutionBuilder.Create()
+                .AddProject(AnalyzerLanguage.CSharp, false)
+                .AddSnippet(@"int number1 = 1;", "MyFile1.cs")
+                .AddSnippet(@"int number2 = 2;", "MyFile2.cs");
```

Comment:
Two files with top-level statements don't compile, right? You may want to fix this.